libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -128,28 +128,12 @@ class AssemblyViewPureChecker
 
 bool ViewPureChecker::check()
 {
-	vector<ContractDefinition const*> contracts;
-
-	for (auto const& node: m_ast)
-	{
-		SourceUnit const* source = dynamic_cast<SourceUnit const*>(node.get());
-		solAssert(source, "");
-		contracts += source->filteredNodes<ContractDefinition>(source->nodes());
-	}
-
-	// Check modifiers first to infer their state mutability.
-	for (auto const& contract: contracts)
-		for (ModifierDefinition const* mod: contract->functionModifiers())
-			mod->accept(*this);
-
-	for (auto const& contract: contracts)
-		contract->accept(*this);
```

Comment:
Where did this block go? Is there a visitor for SourceUnit?

Reply:
SourceUnit is the root AST node and the default visitor just descends into its children